build-system/compile/compile.js

```diff
@@ -372,11 +373,17 @@ function compile(entryModuleFilenames, outputDir, outputFilename, options) {
         .pipe(gulpIf(shouldShortenLicense, shortenLicense()))
         .pipe(sourcemaps.init({loadMaps: true}))
         .pipe(gulpClosureCompile(compilerOptionsArray, distNailgunPort))
+        .pipe(rename(outputFilename))
+        .pipe(
+          gulpIf(
+            !argv.pseudo_names && options.shouldCheckUnknownDeps !== false,
```

Comment:
This logic looks unwieldy. What if you switch `shouldCheckUnknownDeps` to `skipUnknownDepsCheck` with a default value of `false`, and change the `gulpIf` condition to `!argv.pseudo_names && !options.skipUnknownDepsCheck`?